### DISCUSSION TITLE
Simpler asset installation

### DISCUSSION THREAD

**Discussion by xaroth8088** [ikemen-engine/Ikemen-GO#1372] [Feature Request]
### Is your feature request related to a problem? Please describe.

Asset installation and configuration is overly manual, especially when the default options for some/all assets is ok.

### Describe the solution you'd like

I have two potential paths in mind.  Either one or even both could work well to address my use-case.

### Reserved Keyword
Introduce a new reserved keyword, e.g. `autoload`.  At launch, when processing the various `data/.def` files, read the names of all assets inside the relevant well-known directory and expand those into the relevant configuration section.  De-duplicate the assets before proceeding.  Lastly, assume default options for any assets, as though just the name itself had been given.

#### Sample Config
```
[Characters]
autoload
kfmZ, stages/kfm.def
kfm720, stages/kfm.def, order=0
randomselect

[ExtraStages]
autoload
stages/stage1.def
stages/kfm.def
stages/stage0.def
stages/stage0-720.def
```

### Globs
With roughly the same logical steps as above, use GoLang's [Glob](https://pkg.go.dev/path/filepath#Glob) to populate the assets lists.  Because globs can be more fine-grained, assume any options given apply to _all_ matching assets for the given glob (including using a default if no options are given).

#### Sample Config
```
[Characters]
myCustomCharacters/group1/*
myCustomCharacters/group2/*, stages/myGroup2Stage.def
kfmZ, stages/kfm.def
kfm720, stages/kfm.def, order=0
randomselect

[ExtraStages]
my_extra_stages/*/stage-*.def
stages/stage1.def
stages/kfm.def
stages/stage0.def
stages/stage0-720.def
```

### Search terms

assets

**Comment by SuperFromND** [comment]
Another possible idea is, rather than reserving a keyword, introduce some sort of "scanning" feature in the options menu similar to games like [Clone Hero](https://clonehero.net/) or [YARG](https://github.com/YARC-Official/YARG), two games which also revolve primarily around user-created content. Basically you select it and then the game automatically scans their respective content folders, and then populates a list if it finds any valid content. The results of this scan are then cached so you don't have to do it every time you start the game, only when you actually download new stuff.

IKEMEN could probably do something similar since it already has a 'cache' of sorts in the form of `select.def`, and this would also have the side effect of allowing people to still tweak the character order after a scan. And thanks to the flexible menu system, full-game authors could just hide the Scan button from Options entirely, or just ignore it and manually write their `select.def` the old-fashioned way.

**Comment by Fatih120** [comment]
I was about to make a discussion for this, but I agree. Adding characters to the def file manually is something that seems archaic when other games with installable assets have been able to scan them on their own.
